### DISCUSSION TITLE
Metric: Number of iterations

### DISCUSSION THREAD

**Discussion by ottapav** [qpsolvers/qpbenchmark#115] [Design]
It makes sense to me to add a number of iterations as another metric. It does not make sense to compare iterations among the solvers, although it is relevant information for each individual solver itself. What do you think @stephane-caron?

**Comment by stephane-caron** [comment]
Yes, this could be useful for solver developers. Rather than extending the results CSV data for that (the columns wouldn't match per solver), I'd recommend a separate database for each solver with details that are relevant to store (*e.g.* number of iterations, solve/update/polish times for OSQP, etc.).

As long as the data is present in solver return values, it is easy to plug a second database in `run.py` (similarly to what `results` already does, accessing the extra data from the `solution` object). We just want to make sure we do it in a way that extends to multiple solvers (for instance with a base class for solver-specific databases, then separate implementations in child classes).
